tensorflow_datasets/object_detection/deeplesion.py

```diff
@@ -0,0 +1,793 @@
+
+        # build example
+        record = {
+            "image/name": file_name,
+            "image": archive.extract_image(file_name),
+        }
+        if self.builder_config.name == 'abnormal':
+          record.update(bboxs = _format_bboxs(bboxs, im_size))
+        yield idx, record
+
+    elif self.builder_config.name == 'volume':
+      for idx, value in enumerate(split.values):
+        # collect annotations
+        file_name, bboxs, size, ks_idx, m_coords, diameters, n_loc, p_noisy, slice_range, spacing, dicom_windows, p_gender, p_age, sp = value
+
+        # generate name list and cutoff indices for tructated series
+        fileNameUtil = FileNameUtils(file_name)
+        startIdx, endIdx = [int(x) for x in slice_range.split(',')]
+        fname_list = fileNameUtil.get_fname_list(startIdx, endIdx)
+        t_startIdx, t_endIdx = _cal_cutoff_position(
+            len(fname_list), self.builder_config.thickness, ks_idx - startIdx)
+
+        # build example
+        record = {
+            "key_image/name":
+            file_name,
+            "images":
+            archive.extract_images(fname_list[t_startIdx:t_endIdx + 1]),
+            "bboxs":
+            _format_bboxs(bboxs, size),
+            "key_index":
+            ks_idx - startIdx - t_startIdx,
+            "measurement_coord":
+            _format_values(m_coords, 8, float),
+            "lesion_diameters_pixel":
+            _format_values(diameters, 2, float),
+            "normalized_lesion_loc":
+            _format_values(n_loc, 3, float),
+            "possibly_noisy":
+            int(p_noisy),
+            "slice_range": [startIdx, endIdx],
+            "slice_range_trunc":
+            [t_startIdx + startIdx, t_endIdx + startIdx],
+            "spacing_mm_px": [float(x) for x in spacing.split(',')],
+            "image_size": [int(x) for x in size.split(',')],
+            "dicom_windows":
+            _format_values(dicom_windows, 2, int),
+            "patient_gender":
+            int(p_gender),
+            "patient_age":
+            int(p_age),
+        }
+        yield idx, record
+
+    else:
+      raise AssertionError('No builder_config found!')
+
+
+class ArchiveUtils():
+  """Helper class to read image(s) from archives
+
+  Attributes:
+    path: `dict`, paths of archives
+    _lut: `dict`, hash table to lookup file path in archives
+  """
+  def __init__(self, path):
+    """Inits ArchiveUtils with paths of archives"""
+    self.path = path
+    self._lut = None
+
+  @property
+  def lut(self):
+    if not self._lut:
+      self._build_lut()
+    return self._lut
+
+  def extract_image(self, fileName):
+    """Returns a file object according to the fileName
+
+    Args:
+      fileName: `str`, a fileName from the annotation file
+
+    Returns:
+      a file object of the fileName
+    """
+    fileNameUtil = FileNameUtils(fileName)
+    archive = self.lut[fileNameUtil.seriesFolder]
+    with ZipFile(archive, 'r') as zip:
```

Comment:
please use tf.gfile to read the data (so that it also works on non-local systems)